react-validated-input-component/src/index.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react"
+import ReactDOM from "react-dom"
+
+class ValidatedInput extends React.Component{
+  constructor(props){
+    super(props)
+    this.handleChange=this.handleChange.bind(this)
+    this.handleSubmit=this.handleSubmit.bind(this)
+    this.state={password:"", message:"A password is required", icon:"fas fa-times"}
```

Comment:
Well done on this, now the next step is to remove the message and icon properties from state. The reason is that you don't want to store more than what is absolutely necessary in state, and the password is what determines what should be shown on the screen. with that in mind, you can store the extra info as properties on the class so that they are available to the render method